src/HotkeysProvider.tsx

```diff
@@ -30,44 +30,31 @@ interface Props {
 }
 
 export const HotkeysProvider = ({ initiallyActiveScopes = ['*'], children }: Props) => {
-  const [internalActiveScopes, setInternalActiveScopes] = useState(
```

Comment:
Maybe we could have this state wrapped in a React Context to allow separating enableScope and disableScope, thus allowing us to avoid re-renders where we use them. 

This is a very painful problem right now, we cannot call useHotkeysContext() without having re-renders because of enabledScopes changing and causing every component calling useHotkeysContext() to re-render, even if they don't care about this dependency and just want to enable/disable a scope.

Reply:
I would be glad to collaborate on this because we do an extensive usage of scopes at https://github.com/twentyhq/twenty and we have problems managing re-renders.

Reply:
I don't think I quite understand this issue. `enableScope` and `disableScope` are already decoupled.

`enabledScopes` is a state, so changing the value will retrigger depend components. I understand that this causes re renders when calling `enableScope`, but the only fix would be to have the `enabledScopes` in another context, which, API wise, seems weird.

Maybe I am missing something?